### PR TITLE
net-misc/electrum: ebuild improvements

### PR DESCRIPTION
**Pull request by CyberTailor**

### CODE REVIEW

File: net-misc/electrum/electrum-4.2.1.ebuild [hunk 1]
```diff
@@ -7,7 +7,7 @@ DISTUTILS_USE_PEP517=setuptools
 PYTHON_COMPAT=( python3_{8..9} )
 PYTHON_REQ_USE="ncurses?"
 
-inherit desktop distutils-r1 xdg-utils
```
Comment (mgorny) [inline net-misc/electrum/electrum-4.2.1.ebuild:10]: Nope, xdg hate here.

File: net-misc/electrum/electrum-4.2.1.ebuild [hunk 2]
```diff
@@ -51,10 +51,12 @@ BDEPEND="
...
 DOCS=( AUTHORS README.rst RELEASE-NOTES SECURITY.md )
 
+PATCHES=( "${FILESDIR}/3.1.2-no-user-root.patch" )
+
 distutils_enable_tests pytest
 
-src_prepare() {
-	eapply "${FILESDIR}/3.1.2-no-user-root.patch"
+python_prepare_all() {
+	distutils-r1_python_prepare_all
```
Comment (mgorny) [inline net-misc/electrum/electrum-4.2.1.ebuild:59]: Leave it as `src_prepare`, it does the same and it's shorter to type.

File: net-misc/electrum/electrum-4.2.1.ebuild [hunk 3]
```diff
@@ -58,8 +58,8 @@ distutils_enable_tests pytest
 python_prepare_all() {
 	distutils-r1_python_prepare_all
 
-	# Prevent icon from being installed in the wrong location
-	sed -i '/icons_dirname/d' setup.py || die
+	# avoid installing icons and desktop item to wrong location
```
Comment (mgorny) [inline net-misc/electrum/electrum-4.2.1.ebuild:61]: `the` is still needed.

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by CyberTailor** [comment]
Addressed all mentioned issues.

**Comment by blueness** [comment]
Sorry this one fell of my desk.  electrum has now been bumped to 4.2.1-r1.  If you want to update your PR I can apply it.

**Comment by thesamesam** [comment]
@CyberTailor please rebase when you get a chance, thanks

**Pull request closed without merging**